tests/explainers/test_tree.py

```diff
@@ -1841,3 +1841,33 @@ def test_xgboost_dart_regression():
     explainer = shap.TreeExplainer(model)
     shap_values = explainer.shap_values(X)
     assert np.allclose(shap_values.sum(1) + explainer.expected_value, model.predict(X), atol=1e-4)
+
+
+# the expected results can be found in the paper "Consistent Individualized Feature Attribution for Tree Ensembles",
+# https://arxiv.org/abs/1802.03888
+@pytest.mark.parametrize(
+    "expected_result, approximate",
+    [
+        (np.array([[0.0, -20.0], [-40.0, 20.0], [0.0, -20.0], [40.0, 20.0]]), True),
+        (np.array([[-10.0, -10.0], [-30.0, 10.0], [10.0, -30.0], [30.0, 30.0]]), False),
+    ],
+)
+def test_consistency_approximate(expected_result, approximate):
+    """GH #3764.
+    Test that the call interface and shap_values interface are consistent when called with `approximate=True`."""
+
+    import numpy as np
+
+    import shap
+
+    dtc = sklearn.tree.DecisionTreeRegressor(max_depth=2)
+    arr = np.array([[0, 0], [0, 1], [1, 0], [1, 1]])
+    target = np.array([0, 0, 0, 80])
+
+    dtc.fit(arr, target)
+
+    exp = shap.explainers.TreeExplainer(dtc)
+    explanations_call_approx = exp(arr, approximate=approximate)
+    explanations_shap_values_approx = exp.shap_values(arr, approximate=approximate)
+    assert (explanations_call_approx.values == explanations_shap_values_approx).all()
+    assert (explanations_call_approx.values == expected_result).all()
```

Comment:
Would `np.testing.assert_allclose` be more appropriate here?

Reply:
replaced with `np.testing.assert_allclose`